modules/core/ante/ante.go

```diff
@@ -30,10 +32,19 @@ func (rrd RedundantRelayDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 		for _, m := range tx.GetMsgs() {
 			switch msg := m.(type) {
 			case *channeltypes.MsgRecvPacket:
-				response, err := rrd.k.RecvPacket(ctx, msg)
+				var (
+					response *channeltypes.MsgRecvPacketResponse
+					err      error
+				)
+				if ctx.IsReCheckTx() {
+					response, err = rrd.recvPacketReCheckTx(ctx, msg)
+				} else {
+					response, err = rrd.k.RecvPacket(ctx, msg)
```

Comment:
with https://github.com/cosmos/ibc-go/pull/6302 this will be `recvPacketCheckTx`, right? :)